batch/batch/worker/worker.py

```diff
@@ -1494,6 +1494,12 @@ def status(self):
     def done(self):
         return self.state in ('succeeded', 'error', 'failed')
 
+    def log_delayed_warning(self, message, exc_info):
+        last_logged_time_msecs, start_time_msecs = self.last_logged_message[message]
+        if time_msecs() - last_logged_time_msecs >= 300 * 1000:
+            log.warning(message + f' since {time_msecs_str(start_time_msecs)}', exc_info=exc_info)
+            self.last_logged_message[message] = (time_msecs(), start_time_msecs)
```

Comment:
I don't really understand `start_time_msecs` role here. It's never reset, so we just keep printing out the first time this machine ever logged such a warning? I feel like if there's a burst of warnings, a quiet period where everything is fine, and then another burst, the warnings from the second burst have no reason to reference the previous burst.


I feel like this is starting to get pretty complicated. Instead of all this specific bookkeeping, can we instead add some flexibility to retry_transient_errors, where we say something along the lines of "only start logging warnings for X kind of error after X amount of time", and then the only thing we need in worker.py is to say that we should only log warnings for 503s after X minutes?

Reply:
We also should not log 429s. We get 429s when being rate limited and 503s when batch is offline